cmd/launcher/extension.go

```diff
@@ -83,39 +85,19 @@ func createExtensionRuntime(ctx context.Context, db *bbolt.DB, launcherClient se
 		return nil, nil, nil, errors.Wrap(err, "starting grpc extension")
 	}
 
-	// create the logging adapters for osquery
-	osqueryStderrLogger := kolidelog.NewOsqueryLogAdapter(
-		logger,
-		kolidelog.WithLevelFunc(level.Info),
-		kolidelog.WithKeyValue("component", "osquery"),
-		kolidelog.WithKeyValue("osqlevel", "stderr"),
-	)
-	osqueryStdoutLogger := kolidelog.NewOsqueryLogAdapter(
-		logger,
-		kolidelog.WithLevelFunc(level.Debug),
-		kolidelog.WithKeyValue("component", "osquery"),
-		kolidelog.WithKeyValue("osqlevel", "stdout"),
-	)
+	var runnerOptions []runtime.OsqueryInstanceOption
 
-	runner := runtime.LaunchUnstartedInstance(
-		runtime.WithOsquerydBinary(opts.OsquerydPath),
-		runtime.WithRootDirectory(opts.RootDirectory),
-		runtime.WithConfigPluginFlag("kolide_grpc"),
-		runtime.WithLoggerPluginFlag("kolide_grpc"),
-		runtime.WithDistributedPluginFlag("kolide_grpc"),
-		runtime.WithOsqueryExtensionPlugins(
-			config.NewPlugin("kolide_grpc", ext.GenerateConfigs),
-			distributed.NewPlugin("kolide_grpc", ext.GetQueries, ext.WriteResults),
-			osquerylogger.NewPlugin("kolide_grpc", ext.LogString),
-		),
-		runtime.WithOsqueryExtensionPlugins(ktable.LauncherTables(db, opts)...),
-		runtime.WithStdout(osqueryStdoutLogger),
-		runtime.WithStderr(osqueryStderrLogger),
-		runtime.WithLogger(logger),
-		runtime.WithOsqueryVerbose(opts.OsqueryVerbose),
-		runtime.WithOsqueryFlags(opts.OsqueryFlags),
-		runtime.WithAugeasLensFunction(augeas.InstallLenses),
-	)
+	if opts.Transport == "osquery" {
+		var err error
+		runnerOptions, err = osqueryRunnerOptions(logger, db, opts)
```

Comment:
Heh. I think this is one of the warts of golang. I don't like either option, and I don't know what idiomatic is. 

I have a slight preference for what I have, because it makes the scoping on `runnerOptions` more clear.